reframe/core/modules.py

```diff
@@ -729,6 +729,8 @@ def __init__(self):
                 (version, self.MIN_VERSION))
 
         self._version = version
+        self._module_paths_added = []
+        self._module_paths_removed = []
```

Comment:
I would have a single list of tuples here of the form: `[('+', 'path/added'), ('-', '/path/removed')]`. This way you keep the sequence of additions and deletions and be completely sure that you preserve the same behaviour.